scipy/optimize/tests/test_linprog.py

```diff
@@ -802,6 +802,39 @@ def test_bug_5400(self):
                     method=self.method, options=self.options)
             _assert_success(res, desired_fun=-106.63507541835018)
 
+    def test_issue_8174_stackoverflow(self):
+        # Test supplementary example from issue 8174.
+        # https://github.com/scipy/scipy/issues/8174
+        # https://stackoverflow.com/questions/47717012/linprog-in-scipy-optimize-checking-solution
+        c = np.array([1, 0, 0, 0, 0, 0, 0])
+        A_ub = -np.identity(7)
+        b_ub = np.array([[-2],[-2],[-2],[-2],[-2],[-2],[-2]])
+        A_eq = np.array([
+            [1, 1, 1, 1, 1, 1, 0],
+            [0.3, 1.3, 0.9, 0, 0, 0, -1],
+            [0.3, 0, 0, 0, 0, 0, -2/3],
+            [0, 0.65, 0, 0, 0, 0, -1/15],
+            [0, 0, 0.3, 0, 0, 0, -1/15]
+        ])
+        b_eq = np.array([[100],[0],[0],[0],[0]])
+
+        if self.method == 'interior-point':
+            # Warning is raised in presolve method,
+            # which is not yet implemented in the simplex method.
+            with pytest.warns(OptimizeWarning):
+                res = linprog(
+                    c=c, A_ub=A_ub, b_ub=b_ub, A_eq=A_eq, b_eq=b_eq,
+                    method=self.method, options=self.options
+                )
+            _assert_success(res, desired_fun=43.3333333331385)
+
+        elif self.method == 'interior-point':
+            res = linprog(
+                c=c, A_ub=A_ub, b_ub=b_ub, A_eq=A_eq, b_eq=b_eq,
+                method=self.method, options=self.options
+            )
+            _assert_success(res, desired_fun=43.3333333331385)
```

Comment:
@Kai-Striega Why not just one of these outside the `if-elif`? 
But I suppose If I see the tests pass tonight I'll merge.

Reply:
If added this using if-elseif to avoid having the test run if an invalid test is run.
However having the test run outside of the if block raises an unbound local error.

Reply:
I don't understand what you mean by:
"If added this using if-elseif to avoid having the test run if an invalid test is run."

It sounds like by "invalid test" you mean if `self.method` takes on some value other than `simplex` or `interior-point`. That will not happen until we add a new method, and then we'll have to review this test anyway. It would probably be better if this caused the test to fail, actually, because it reminds us that we need to reconsider how this test should be applied for the new method.

Also, we're going to adding presolve to `simplex` soon, right? So soon enough, we'll actually want to take out the if/elif altogether, because the test should be applied to both methods the same way. I don't anticipate that we'll be adding any new methods before then.

Whether you use this `if-elif` or `if-else` like I suggest, `res` is guaranteed to be defined as the tests are currently written, so I don't see the harm of putting a single `_assert_success` outside these blocks.

I don't see how an `UnboundLocalError` is relevant, as no assignments are made to `res` (see [here](https://eli.thegreenplace.net/2011/05/15/understanding-unboundlocalerror-in-python) for an explanation `UnboundLocalError`).

I suggest trying my suggestion out. If you do get some error, please let me know, but if the tests pass with a single `_assert_success` outside these blocks, I think that's how it should be.